pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1504,14 +1545,58 @@ func isDeviceOpened(deviceToDetach AttachedVolume, mounter mount.Interface) (boo
 	return deviceOpened, nil
 }
 
-func isMigrated(vpm *volume.VolumePluginMgr, spec *volume.Spec) bool {
+func isNodeMigrated(og *operationGenerator, pluginName string, nodeName types.NodeName) (bool, error) {
+	var err error
+	var nodeInfo *csiapi.CSINodeInfo
+
+	if og.csiNodeInfoHasSynced() && utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) {
```

Comment:
The `utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo)` can be moved out and the checks structured like: 
```
if utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) {
    if og.csiNodeInfoHasSynced() {
        ...
    } else {

    }
} else {
    return false, fmt.Errorf("failed to check if node migrated, CSINodeInfo feature gate not enabled")
}
```